### PR TITLE
map database mainpage + creating link to category

### PR DESCRIPTION
**Pull request by ostermann-felix**

### CODE REVIEW

File: components/CategoryGrid.js [hunk 1]
```diff
@@ -0,0 +1,54 @@
+import styled from 'styled-components';
+import Link from 'next/link';
+import Image from 'next/image';
+
+export default function CategoryGrid({ lessons }) {
+  console.log(lessons);
```
Comment (NikRoe) [inline components/CategoryGrid.js:6]: you can remove this console.log

File: components/CategoryGrid.js [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+  return (
+    <CatGrid>
+      {lessons &&
+        lessons.map((lesson) => (
+          <IconBox key={lesson.id}>
+            <Link href={`/categories/${lesson.category.name}`} passHref>
+              <CatLink>
```
Comment (NikRoe) [inline components/CategoryGrid.js:13]: is this a link to a cat? 🤔 🐱 

how about `CategoryLink`?

File: components/Footer.js [hunk 3]
```diff
@@ -1,13 +1,11 @@
...
 import styled from 'styled-components';
-import backwardicon from '../Assets/Icons/backward.ico';
-import forwardicon from '../Assets/Icons/forward.ico';
 
 export default function Footer() {
   return (
     <>
       <Footerbody>
-        <Backward
+        {/* <Backward
```
Comment (NikRoe) [inline components/Footer.js]: you can remove the comment and the parts within

File: components/PlaceholderImg.js [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+const ImgBackground = styled.div`
+  height: 20vh;
+  display: flex;
+  align-items: center;
+  justify-content: center;
+  border-radius: 12px;
+  box-shadow: rgba(100, 100, 111, 0.2) 0px 7px 29px 0px;
+`;
+
+const Little_paramedic = styled.div``;
```
Comment (NikRoe) [inline components/PlaceholderImg.js]: you can remove the styled component as it is not setting any CSS properties and use a simple `div` instead

File: pages/categorie.js [hunk 5]
```diff
@@ -13,7 +13,6 @@ export async function getServerSideProps() {
 function Categorie({ lessons }) {
```
Comment (NikRoe) [inline pages/categorie.js]: the page should be called `Category` and the page it resides in `pages/category.js`

File: pages/categories/[category].js [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+  const lessons = await getCategoryByName(category);
+  return {
+    props: {
+      lessons: lessons,
+      category: category,
+    },
+  };
+}
+
+export default function lessonPage({ category, lessons }) {
```
Comment (NikRoe) [inline pages/categories/[category].js]:
```suggestion
export default function LessonPage({ category, lessons }) {
```

File: pages/categories/[category].js [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+      </CategoryWrapper>
+      {lessons.map((lesson, index) => {
+        return (
+          <>
+            <LessonBox>
+              <Link href={`/lessons/${lesson.id}`} alt="" key={lesson.id}>
+                {lesson.name}
+              </Link>
+            </LessonBox>
+          </>
```
Comment (NikRoe) [inline pages/categories/[category].js]:
```suggestion
            <LessonBox key={lesson.id}>
              <Link href={`/lessons/${lesson.id}`}>
                {lesson.name}
              </Link>
            </LessonBox>
```

File: pages/lessons/[id].js [hunk 8]
```diff
@@ -11,19 +13,33 @@ export async function getServerSideProps(context) {
 }
 
 export default function lessonPage({ lesson }) {
-  console.log(lesson, 'Banane');
-
   return (
     <>
       <h2>{lesson.name}</h2>
       {lesson.process.map((process, index) => {
+        console.log(process.visual);
```
Comment (NikRoe) [inline pages/lessons/[id].js]: you can remove this console.log

File: pages/lessons/[id].js [hunk 9]
```diff
@@ -11,19 +13,33 @@ export async function getServerSideProps(context) {
...
+            {process.visual ? (
+              <LessonVisual>
+                <Image
+                  key={`${process.name} ${index}`}
+                  alt="instruction visual"
+                  src={process.visual}
+                  width={350}
+                  height={160}
+                />
+              </LessonVisual>
```
Comment (NikRoe) [inline pages/lessons/[id].js]:
```suggestion
              <LessonVisual key={`${process.name} ${index}`}>
                <Image
                  alt="instruction visual"
                  src={process.visual}
                  width={350}
                  height={160}
                />
              </LessonVisual>
```

**Review by NikRoe — CHANGES_REQUESTED** [review]
Hey Felix,

looking good for far, I left a few remarks considering console.logs and namings etc.

Applying those suggestions will make your app even more clean.

File: components/Footer.js (hunk 3 above)
Comment (ostermann-felix) [reply components/Footer.js]: Done!

File: components/PlaceholderImg.js (hunk 4 above)
Comment (ostermann-felix) [reply components/PlaceholderImg.js]: Done!

File: pages/categorie.js (hunk 5 above)
Comment (ostermann-felix) [reply pages/categorie.js]: Deleted

File: pages/categories/[category].js (hunk 6 above)
Comment (ostermann-felix) [reply pages/categories/[category].js]: Changed to CategoryPage

File: pages/categories/[category].js (hunk 7 above)
Comment (ostermann-felix) [reply pages/categories/[category].js]: Done!

File: pages/lessons/[id].js (hunk 8 above)
Comment (ostermann-felix) [reply pages/lessons/[id].js]: Done!

File: pages/lessons/[id].js (hunk 9 above)
Comment (ostermann-felix) [reply pages/lessons/[id].js]: Done!

**Review by NikRoe — APPROVED** [review]
Well done Felix 🙌

**Pull request merged**